mktestimg.c

```diff
@@ -22,87 +22,123 @@ enum {
     FORMAT_COUNT,
 };
 
-char *copystr(const char *string) {
-    long len = strlen(string); 
-    char *out = malloc(sizeof(char) * (len + 1));
-    strcpy(out, string); 
-    out[len] = '\0';
+char* copy_str(const char* string) {
+    if (string == NULL) {
```

Comment:
This check shouldn't be here. I know you concern that this causes a segfault:
``./mktestimg -w 10 -h 10 -c 4 -f 1 -o`` but this check instead should be present [here](https://github.com/iddev5/mktestimg/blob/master/mktestimg.c#L69) ``if (argv[i]) output = copystr(argv[i])``

Reply:
It's a bad idea to handle things like that outside of the function itself. Make things foolproof or you will spend even more time with bugs.

Reply:
It is full proof. copystr("out.img"), which is the only other call of copystr() is a const char*, it cannot be NULL.

Reply:
*The parameter string being NULL isn't actually an issue* – this is tremendously misleading and tends to longer arguments. In every step, nevertheless, the chances of issues being created are negligible – it's the programmers' responsibility to verify whether the values passed, assigned, or manipulated are valid or not, which might seem like writing some redundant code.

Reply:
I agree with rohanbari here.

Reply:
There seem to be some misunderstanding or maybe I didnt worded it correctly. Let me explain it in detail.

Basically, the argument parser I made is very basic and its quite fallable. Every ``argv[i]`` can fail for ``i >= argc`` which is possible when we do ``atoi(argv[++i])``. Now in this case the problem is not visible because ``atoi(NULL)`` returns 0. But this does causes problem with copying of string. So my point was that, the check should be done globally for everyone, not just for the one case of ``-o``. 

Like if i input ``./mktestimg -w``, it shows error that width or height should be greater than 0. But in reality it should give an error like ``arguments are incomplete``. The same should also be designed for the ``-o`` parameter. 

But, again, in order to clear all your doubts/concerns, I found a middle way :tada: Basically I am getting rid of all this memory allocation thing in general. Just directly assigning the address of ``argv[i]`` to ``char *output`` and in case it is null, it is later checked and then reassigned by ``static char *out_filename = "out.img";``. This totally eliminates the problem of memory as we are dealing with static const char slices here. It will get rid of the whole ``copystr()`` function. So its all safe. So peace now? :v: 

(I will link the commit once its done, it would make more sense with code)